File: iteration2/src/main/webapp/resources/js/service/OrderService.js
/**
 * Created by Colota on 4/23/2017.
 */
'use strict';

angular.module('OrderModule')
    .service('OrderService',['$resource',OrderService]);

function OrderService($resource) {
    return $resource(
        'http://localhost:8080/api/order/:id',
        {id: '@id'},//Handy for update & delete. id will be set with id of instance
        {
            update: {
                method: 'PUT' // To send the HTTP Put request when calling this custom update method.
            }

        }
    );
}

File: iteration2/src/main/java/com/dynamicheart/bookstore/service/OrderService.java
package com.dynamicheart.bookstore.service;

import com.dynamicheart.bookstore.domain.Order;

import java.util.List;

/**
 * Created by dynamicheart on 4/23/2017.
 */
public interface OrderService {
    Order findOne(Long id);
    List<Order> findOrders();
    void update(Order order);
    void save(Order order);
    void delete(Long id);
    boolean isExist(Order order);
}
